### PR TITLE
feat: Allow closeButton color in DismissableBanner to be configurable

### PR DESCRIPTION
**Pull request by Hennnnes**
## Hey, I just made a Pull Request!

I updated the default link and button color inside of the `DismissableBanner` so they now match the text color if they are not specified. Furthermore I added the possibility to configure the color of the button via `theme.palette.banner.button`.

**Before**
<table>
<tr>
  <td>
      Lightmode
  </td>
  <td>
     Darkmode
  </td>
</tr>
<tr>
  <td>image</td>
  <td>
image
</td>
</tr>
</table>

**After**
<table>
<tr>
  <td>
      Lightmode
  </td>
  <td>
     Darkmode
  </td>
</tr>
<tr>
  <td>image</td>
  <td>
image
</td>
</tr>
</table>

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/lucky-badgers-own.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/core-components': minor
+'@backstage/theme': minor
+---
+
+Update default color of DismissableBanner link and button and make button color configurable
```
Comment (awanlin) [inline .changeset/lucky-badgers-own.md:6]: Can you include an example here, please? The changesets get turned into release note so having an example will help people as they upgrade

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for the contribution @Hennnnes. Just left one comment and then have some general questions/feedback.

Backstage uses Storybook so we should be updating it with examples, there is one for this component so we should update it: https://backstage.io/storybook/?path=/story/feedback-dismissablebanner--default

Now, about the actual change. I'm not seeing how the button change would look, can you include an example, please? I do find that having the text all be the same color makes the link less obvious. Will this change be opt-in or will everyone be forced into this?

**Comment by Hennnnes** [comment]
Hey @awanlin,
thanks for your feedback and sorry for the late reply (I was on vacation).

> I do find that having the text all be the same color makes the link less obvious.
The possibility to change the color of the link is untouched, but with my current changes indeed the change would not be opt-in. I'll refactor the code and maybe remove the `inherit` color of the link again, as I now see that the wanted default behaviour is a different color and not the same as the text :)

Sure, I'll include an example in the changelog and update the storybook file :)

**Comment by Hennnnes** [comment]
@awanlin I just now realised, that my naming was probably not chosen that well. I updated `button` to `closeButton` as I wanted to be able to customise the color of the close button on the banner, not the color of an actual button in the banner. Please also take a look at the updated screenshots in the PR description.

Therefore I guess there is no need to update any stories, right?

**Review by awanlin — APPROVED** [review]
Thanks again @Hennnnes this looks good to me

Now we just need @benjdlambert to review

File: packages/theme/api-report.md [hunk 2]
```diff
@@ -70,6 +70,7 @@ export type BackstagePaletteAdditions = {
     error: string;
     text: string;
     link: string;
+    closeButtonColor: string;
```
Comment (benjdlambert) [inline packages/theme/api-report.md:73]: Wonder if we should make this optional and have a sensible fallback, that way we don't break typescript for everyone.

File: .changeset/shiny-crabs-flow.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/core-components': minor
+'@backstage/theme': minor
```
Comment (benjdlambert) [inline .changeset/shiny-crabs-flow.md:3]:
```suggestion
'@backstage/core-components': patch
'@backstage/theme': patch
```

File: .changeset/shiny-crabs-flow.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+---
+
+Allow closeButton color in DismissableBanner to be configurable (via. theme.palette.banner.closeButtonColor)
```
Comment (benjdlambert) [inline .changeset/shiny-crabs-flow.md:6]:
```suggestion
Allow` closeButton` color in `DismissableBanner` to be configurable (via. `theme.palette.banner.closeButtonColor`)
```

**Review by benjdlambert — CHANGES_REQUESTED** [review]

File: packages/theme/api-report.md (hunk 2 above)
Comment (Hennnnes) [reply packages/theme/api-report.md:73]: Yeah, I guess that will work out better. I'll add it as an additional commit to get your feedback on it :)

**Review by benjdlambert — APPROVED** [review]
Thanks! :pray:

**Pull request merged**